Sources/SemanticAnalyzer/TypeChecker.swift

```diff
@@ -297,7 +297,28 @@ public struct TypeChecker: ASTPass {
   }
 
   public func postProcess(subscriptExpression: SubscriptExpression, passContext: ASTPassContext) -> ASTPassResult<SubscriptExpression> {
-    return ASTPassResult(element: subscriptExpression, diagnostics: [], passContext: passContext)
+    var diagnostics = [Diagnostic]()
+    let environment = passContext.environment!
+    let typeIdentifier = passContext.enclosingTypeIdentifier!
+    let scopeContext = passContext.scopeContext!
+
+    let identifierType = environment.type(of: .identifier(subscriptExpression.baseIdentifier), enclosingType: typeIdentifier.name, scopeContext: scopeContext)
+
+    var expectedType: Type.RawType;
```

Comment:
Oh no! A wild semi-colon appeared 👻

Reply:
Old habits die hard! Cheers for picking it up